service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/PlacementController.java

```diff
@@ -85,13 +87,27 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             .setChild(child);
 
         caseProperties.put("placements", placementService.setPlacement(caseData, placement));
-        removeTemporaryFields(caseDetails, "placement", "placementChildName", "singleChild");
+        removeTemporaryFields(caseDetails, "placementChildName", "singleChild");
 
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseProperties)
             .build();
     }
 
+    @PostMapping("/submitted")
+    public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
+        var placement = mapper.convertValue(callbackRequest.getCaseDetails().getData().get("placement"),
```

Comment:
👀 look at you being all java 10 😆

Reply:
Can't wait to use fancy java 12+ switch syntax one day

Reply:
Soon, also the text blocks in java 13 and [this in java 14](https://openjdk.java.net/jeps/358)